packages/txstream/tangleledger/tangleledger.go

```diff
@@ -71,13 +71,18 @@ func (t *TangleLedger) GetUnspentOutputs(addr ledgerstate.Address, f func(output
 
 // GetHighGoFTransaction fetches a transaction by ID, and executes the given callback if its GoF is high.
 func (t *TangleLedger) GetHighGoFTransaction(txid ledgerstate.TransactionID, f func(ret *ledgerstate.Transaction)) (found bool) {
-	found = false
+	found = true
 	t.tangleInstance.LedgerState.TransactionMetadata(txid).Consume(func(txmeta *ledgerstate.TransactionMetadata) {
-		if t.tangleInstance.ConfirmationOracle.IsBranchConfirmed(txmeta.BranchID()) {
-			found = true
-			t.tangleInstance.LedgerState.Transaction(txid).Consume(f)
+		for txBranchID := range txmeta.BranchIDs() {
+			if found = found && t.tangleInstance.ConfirmationOracle.IsBranchConfirmed(txBranchID); !found {
+				break
+			}
 		}
 	})
+
+	if found {
```

Comment:
shouldnt it only execute the callback if ALL branches have high gof AND the tx has high gof?